contracts/peripheral/grants/VeBalGrant.sol

```diff
@@ -0,0 +1,352 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import { IBalancerVault, IPriceOracle, IAsset } from "../../interfaces/balancer/IBalancerCore.sol";
+import { IFeeDistributor } from "../../interfaces/balancer/IFeeDistributor.sol";
+import { IBalGaugeController } from "../../interfaces/balancer/IBalGaugeController.sol";
+import { IVotingEscrow } from "../../interfaces/balancer/IVotingEscrow.sol";
+import { AuraMath } from "../../utils/AuraMath.sol";
+import { IERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title   VeBalGrant
+ * @author  AuraFinance
+ * @notice  An escrow contract for the BAL grant provided to projects
+ * @dev     Allows projects
+ */
+contract VeBalGrant {
+    using SafeERC20 for IERC20;
+    using AuraMath for uint256;
+
+    /* ----------------------------------------------------------------
+       Storage 
+    ---------------------------------------------------------------- */
+
+    IERC20 public immutable WETH;
+    IERC20 public immutable BAL;
+    IERC20 public immutable BAL_ETH_BPT;
+    IVotingEscrow public immutable votingEscrow;
+    IBalGaugeController public immutable gaugeController;
+    address public immutable project;
+    address public immutable balancer;
+    bool public active;
+    bool public hasLock;
```

Comment:
Nit: I would reorder these so that immutable vars are all together then it matches the order of the constructor vars.